synapseclient/multipart_upload.py

```diff
@@ -295,8 +294,28 @@ def _upload_chunk(part, completed, status, syn, filename, get_chunk_function,
             sys.stderr.write(str(ex1))
             sys.stderr.write("Encountered an exception: %s. Retrying...\n" % str(type(ex1)))
 
+class DummyPool():
```

Comment:
Since this is a temporary workaround until AWS Lambda gets fixed we should try to make as little change to the current multipart_upload as possible.

Move the Dummy* classes into own module file. (I also recommend not having "dummy" in the module name since that might be confused with multiprocessing.dummy)

rename these classes to be Pool() and Value() and make their __init__() emulate that of the multiprocessing module classes. 
e.g. you can do fakemultiprocessing.Pool(8) but the 8 gets ignored. ('fakemultiprocessing' is just a placeholder for a module name that you think would fit)

then in the import section you can do a try block:
```
try:
   from multiprocessing import Value
   from multiprocessing.dummy import Pool
   Value('d',0) # just try to create a value and it should fail on AWS Lambda; no need to assign to a variable
except OSError:
   from .fakemultiprocessing import Pool, Value
```
Now there is no need for the if check on the `singleThreaded ` boolean and the unchanged logic in multipart_upload should work as is